### ISSUE TITLE
Navigation.push in sideMenu doesn't respect bottomTabs visibility options

### ISSUE THREAD

**Issue by markhomoki** [wix/react-native-navigation#6989]
## 🐛 Bug Report
Cannot hide bottomTabs when pushing to a new Bottom Tabs Screen from Side Menu.
Using Navigation.push() from any other Bottom Tabs screens passes `bottomTabs: { visible: false }` but when calling from a Side Menu screen it does not.
This used to work in v4, but started migrating to v7 this week and blocked by this issue.

Any help or pointing to the right direction is appreciated. 

### Have you read the [Contributing Guidelines on issues](https://wix.github.io/react-native-navigation/docs/meta-contributing/)?
Yes

## To Reproduce
```
Navigation.setRoot({
  root: {
    sideMenu: {
      left: {
        component: {
          id: 'SideMenu',
          name: 'Menu'
        }
      },
      center: {
        bottomTabs: {
          children: [{
            stack: {
              children: [{
                component: {
                  id: 'HOME_SCREEN',
                  name: 'Home'
                }
              }],
              options: {
                bottomTab: {
                  text: 'Home',
                  icon: require('./home.png')
                }
              }
            }
          }]
        }
      }
    }
  }
});

// calling this code from Side Menu won't apply visible: false
Navigation.push('HOME_SCREEN', {
  component: {
    name: 'Settings',
    options: {
      bottomTabs: {
        visible: false
      }
    }
  }
});
```

Tried to debug the native code and I believe the issue is related to `topMostViewController`,
which got called in `ReactNativeNavigation/RNNBasePresenter.m` `hidesBottomBarWhenPushed`.

In that function, changing `self.boundViewController.topMostViewController.resolveOptions` to `self.boundViewController.resolveOptions` fixes the issue, but I'm afraid that it will have side effects.

When pushing inside Bottom Tabs from Bottom Tabs:
- topMostViewController.resolveOptions has a value of `bottomTabs.visible: NO` and
- boundViewController.resolveOptions has a value of `bottomTabs.visible: NO`

When pushing inside Bottom Tabs from Side Menu:
- topMostViewController.resolveOptions has a value of `bottomTabs.visible: nil` while
- boundViewController.resolveOptions has a value of `bottomTabs.visible: NO`

Because `bottomTabs.visible: nil`, the code relies on `self.defaultOptions` and during merge it sets `bottomTabs.visible: YES`.

## Expected behavior
When using Navigation.push method should respect `options` property.

## Actual Behavior
Passing `options` property to Navigation.push doesn't get applied when push happens from Side Menu screen.

## Your Environment

- React Native Navigation version: 7.11.0
- React Native version: 0.63.3
- Platform(s) (iOS, Android, or both?): iOS
- Device info (Simulator/Device? OS version? Debug/Release?): 14.4 Simulator and Device

**Comment by yogevbd** [comment]
This looks like an iOS bug where the system doesn't take `hidesBottomBarWhenPushed` into consideration when the first `UITabBarController` child is different than `UINavigationController`. Managed to reproduce it in a plain iOS project.
I think that our best option is to add a workaround that will manually hide the `bottomTabs` in this specific case.

**Comment by markhomoki** [comment]
What would the best way to hide manually in that case? I tried to apply static options which didn't hide it. When I used Navigation.mergeOptions then it was a flash and the UX wasn't the same as navigating from a Bottom Tab screen.

Would it be an option to check somehow whether it's from sideMenu or not inside `hidesBottomBarWhenPushed` and use `self.boundViewController.resolveOptions` when navigating from sideMenu?

**Comment by markhomoki** [comment]
@yogevbd Is there anything you can recommend us looking into and potentially apply a fix? Upgrading Android fixed lot of issues for us and now we are maintaining 2 branches, which is not ideal. One for Android with upgraded RNN, and another one for iOS with the old RNN.

Any suggestions are welcome. We will try to allocate some time next week to solve this.

**Comment by rocketplayfever** [comment]
Same issue in 7.37.1
Any workarounds?

**Comment by vitalyster** [comment]
After hours of bisecting, I finally found problematic changes:
1. The last working good version is `6.4.0`.
2. [This PR](https://github.com/wix/react-native-navigation/pull/6095) introduces the regression

Right now I'm ended up using 6.4.0 + [safeareaview fixes](0da097ef8471670e6550152fa5ebbdf4a02b3478) backport and all issues gone for me. But it could be great if someone will be able to fix the regression.

**Comment by vitalyster** [comment]
Found a workaround using **latest library version**: navigation overlay accepts alternative stack which perfectly displayed on top of root stack with bottom tabs.

1. When user pressed item in side menu use: 
```js
Navigation.showOverlay({
  stack: {
    children: [{
      id: 'MenuItemScreen'
    }],
    options: {
     topBar: {
       rightButtons: [{
         id: 'DismissButton',
         icon: require('dismiss.png')
       }]
     }
    }
  }
})
```
2. If you need to push next screens use this component id:
```js
Navigation.push('MenuItemScreen', { ... })
```
3. Unfortunately this trick does not work on Android, it throw "stack MenuItemScreen not found" error on push. So you need to use this trick conditionally:
```js
// create overlay or push
if (Platform.OS === 'ios') {
  Navigation.showOverlay({ ... })
} else {
  Navigation.push('bottomTabsComponentId', { ... })
}
// push next screen
const stackOwner = Platform.OS === 'ios' ? 'MenuItemScreen' : 'bottomTabsComponentId'
Navigation.push(stackOwner, { ... })
```

4. To dismiss overlay:
```js
Navigation.events().registerNavigationButtonPressedListener(({ buttonId }) => {
    if (buttonId === 'DismissButton') {
      Navigation.dismissAllOverlays()
    }
  })
```

**Comment by yedidyak** [comment]
Closing this as part of a maintainer cleanup of open issues older than one year.

Decision: Closed: stale/non-critical; no evidence it is a critical current-version RNN issue.

If this still reproduces on the latest React Native Navigation with a minimal reproduction, please open a new issue with current RN/RNN versions and repro steps.

**Comment by vitalyster** [comment]
Still reproduced
